tapscript/tx.go

```diff
@@ -194,6 +219,16 @@ func VirtualTx(newAsset *asset.Asset, prevAssets commitment.InputSet) (
 		err       error
 	)
 
+	// Check that we support the provided sighash flag.
+	switch sighashFlag {
```

Comment:
Same here, sighash can be about the signing, not affect the tx construction persay.

Reply:
Moved this check to the VM validation level